packages/block-editor/src/hooks/grid-visualizer.js

```diff
@@ -4,52 +4,65 @@
 import { createHigherOrderComponent } from '@wordpress/compose';
 import { addFilter } from '@wordpress/hooks';
 import { useSelect } from '@wordpress/data';
+import { createContext, forwardRef, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { GridVisualizer, useGridLayoutSync } from '../components/grid';
 import { store as blockEditorStore } from '../store';
 
+export const GridResizerBoundsContext = createContext();
+
 function GridLayoutSync( props ) {
 	useGridLayoutSync( props );
 }
 
-function GridTools( { clientId, layout } ) {
-	const { isSelected, isDragging } = useSelect( ( select ) => {
-		const { isBlockSelected, isDraggingBlocks } =
+const GridTools = forwardRef( ( { clientId, layout }, ref ) => {
+	const { hasSelection, isDragging } = useSelect( ( select ) => {
+		const { isBlockSelected, hasSelectedInnerBlock, isDraggingBlocks } =
 			select( blockEditorStore );
 
 		return {
-			isSelected: isBlockSelected( clientId ),
+			hasSelection:
+				isBlockSelected( clientId ) ||
+				hasSelectedInnerBlock( clientId ),
 			isDragging: isDraggingBlocks(),
 		};
 	} );
 
 	return (
 		<>
 			<GridLayoutSync clientId={ clientId } />
-			{ ( isSelected || isDragging ) && (
-				<GridVisualizer clientId={ clientId } parentLayout={ layout } />
+			{ ( hasSelection || isDragging ) && (
+				<GridVisualizer
+					clientId={ clientId }
+					parentLayout={ layout }
+					contentRef={ ref }
+				/>
 			) }
 		</>
 	);
-}
+} );
 
 const addGridVisualizerToBlockEdit = createHigherOrderComponent(
 	( BlockEdit ) => ( props ) => {
+		// Use useState() instead of useRef() so that GridItemResizer updates when ref is set.
+		const [ resizerBounds, setResizerBounds ] = useState();
```

Comment:
Maybe move this `useState` and the `Provider` to a new component so that we're not calling `useState` on every single block regardless of whether it has a grid layout. Probably doesn't make any difference to performance in practice but I think it's a nice habit to be really careful about what goes above the early return in a `BlockEdit` filter.

Reply:
Good point, I think it's also much tidier to move it into `GridTools` and have `GridTools` accept `children`. Done in 6139488